### ISSUE TITLE
Authentication fails when username and password contain hyphens

### ISSUE THREAD

**Issue by simondean** [prisma/prisma#16370]
Hi.  I'm trying to use Prisma in a situation where a AWS RDS MySQL username and password both contain hyphens.  I've found Prisma always fails to connect when the username and password contain hyphens.  It fails with this exception: 

> the provided database credentials for `(not available)` are not valid

The `(not available)` bit is actually what is included in the error message - it is not me obscuring part of the error message.  That `(not available)` in the error message seems to be caused by the username containing hyphens

**Comment by miguelff** [comment]
If you only change the password to not contain hyphen, Does the same error occur?

If you only change the user to not contain hyphen, while keeping the same password containing a hyphen. Does the same error occur?

Internal note: besides understanding if we are tampering with the username; or MySQL is rejecting it, and we can do something about it. Maybe the error can be made more obvious?

**Comment by janpio** [comment]
Can you share an anonymized version of the credentials you are using @simondean? Just replace all the characters with other ones or something, I would love to see the "structure" of what you are using there that causes this error. Maybe that would already be enough for us to reproduce as well.

**Comment by jharrell** [comment]
Hello again, 

Without further info, we'll need to close this issue. If you are still experiencing this issue, please leave a comment or open a new issue. Thank you!

**Comment by simondean** [comment]
Hi @Weakky, @jharrell.  Sorry I've not replied sooner.  The component I was writing when I hit the issue, I had to switch to `mysql2` package due to time frames for implementing it, so I've not got an immediate need for the fix.  

> If you only change the password to not contain hyphen, Does the same error occur?
> If you only change the user to not contain hyphen, while keeping the same password containing a hyphen. Does the same error occur?

I'm not sure sorry.  I wasn't in control of the user name or password, so I wasn't able to change either to see if the root cause was with user names or passwords

I think the issue I saw was `prisma` was escaping the hyphens in the user name and/or password (not sure which) - could have been percent encoding but not sure - when the escaping was not needed.  The escaping was actually breaking things
